src/auth/computeclient.ts

```diff
@@ -80,13 +80,16 @@ export class Compute extends OAuth2Client {
     const tokenPath = `service-accounts/${this.serviceAccountEmail}/token`;
     let data: CredentialRequest;
     try {
-      data = await gcpMetadata.instance({
+      const instanceOptions: gcpMetadata.Options = {
         property: tokenPath,
-        params: {
-          scopes: this.scopes,
-          // TODO: clean up before submit, fix upstream type bug
-        } as {},
-      });
+      };
+      if (this.scopes.length > 0) {
+        instanceOptions.params = {
+          scopes: this.scopes.join(','),
+        };
+        // TODO: clean up before submit, fix upstream type bug
```

Comment:
do we need this TODO comment? don't quite follow it.

Reply:
Carried over from @JustinBeckwith, who might have intended to fix the problem this PR solves before it shipped, but forgot 🤷‍♂. Interested to see if he has a better fix in mind!

Reply:
You're change resolves the TODO (modulo the 'upstream bug' part). It can be removed.

Reply:
What upstream type bug, though? Did Justin want to change the way params were parsed upstream? I guess I would like to make sure before removing the comment or merging this PR, as the clarity could nullify either idea.

Reply:
At one point there was a cast here, so that can be removed 🙃